src/_pytest/nodes.py

```diff
@@ -254,13 +261,13 @@ def getparent(self, cls):
     def _prunetraceback(self, excinfo):
         pass
 
-    def _repr_failure_py(self, excinfo, style=None):
-        # Type ignored: see comment where fail.Exception is defined.
-        if excinfo.errisinstance(fail.Exception):  # type: ignore
+    def _repr_failure_py(
+        self, excinfo: ExceptionInfo[Union[Failed, "FixtureLookupError"]], style=None
```

Comment:
Are we sure only `Failed` and `FixtureLookupError` are possible here?

Reply:
That are the ones I've found - can be adjusted if needed?

Reply:
It depends on what the intention of the function is - to accept only these two types, or to accept any `ExceptionInfo[E]`.

Reply:
My intention is to being made aware if other types are used.
This should aid with refactoring / cleanup - my main motivation to add types here.